mobile/android/android-application-manager.ts

```diff
@@ -44,10 +44,22 @@ export class AndroidApplicationManager extends ApplicationManagerBase {
 	}
 
 	public async startApplication(appIdentifier: string): Promise<void> {
-		await this.adb.executeShellCommand(["monkey",
-			"-p", appIdentifier,
-			"-c", "android.intent.category.LAUNCHER",
-			"1"]);
+		const pmDumpOutput = await this.adb.executeShellCommand(["pm", "dump", appIdentifier, "|", "grep", "-A", "1", "MAIN"]);
+		const fullActivityNameRegExp = this.getFullyQualifiedActivityRegex();
+		const activityMatch = new RegExp(fullActivityNameRegExp, "m");
```

Comment:
I do need the "m" flag because we are searching for "MAIN", but the actual string we're matching is on the next line.